google/cloud/storage/async/client.h

```diff
@@ -252,6 +252,66 @@ class AsyncClient {
          std::move(options)});
   }
 
+  /**
+   * Uploads a new object without buffering.
+   *
+   * Use this API to upload objects without any client-side buffering. The API
+   * permits resuming an upload after a transient error. If the application
+   * can checkpoint a simple string the API can even be used to resume an upload
+   * even after the application restarts.
```

Comment:
If the application can checkpoint a simple string, the API can be used to resume an upload, even after the application restarts.

Reply:
Done